backend/src/media/backends/s3-backend.ts

```diff
@@ -25,23 +25,24 @@ export class S3Backend implements MediaBackend {
     private mediaConfig: MediaConfig,
   ) {
     this.logger.setContext(S3Backend.name);
-    if (mediaConfig.backend.use === BackendType.S3) {
-      this.config = mediaConfig.backend.s3;
-      const url = new URL(this.config.endPoint);
-      const secure = url.protocol === 'https:'; // url.protocol contains a trailing ':'
-      const endpoint = `${url.hostname}${url.pathname}`;
-      let port = parseInt(url.port);
-      if (isNaN(port)) {
-        port = secure ? 443 : 80;
-      }
-      this.client = new Client({
-        endPoint: endpoint.substr(0, endpoint.length - 1), // remove trailing '/'
-        port: port,
-        useSSL: secure,
-        accessKey: this.config.accessKeyId,
-        secretKey: this.config.secretAccessKey,
-      });
+    if (mediaConfig.backend.use !== BackendType.S3) {
+      return;
     }
+    this.config = mediaConfig.backend.s3;
+    const url = new URL(this.config.endPoint);
+    const isSecure = url.protocol === 'https:';
+    this.client = new Client({
+      endPoint: url.hostname,
+      port: this.determinePort(url),
+      useSSL: isSecure,
+      accessKey: this.config.accessKeyId,
+      secretKey: this.config.secretAccessKey,
+    });
+  }
+
+  private determinePort(url: URL): number | undefined {
+    const port = parseInt(url.port);
+    return isNaN(port) ? undefined : port;
   }
```

Comment:
Really simple method, but for the coverage a unit-test would be good

Reply:
It's a private method. Therefore i would need to add unit tests for the whole class. That's not a bad thing, but could take some more time.

Reply:
I've added a unit test for the s3-backend